lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2761,6 +2828,11 @@ bool LLVMToSPIRVBase::transBuiltinSet() {
             SPIRVBuiltinSetNameMap::map(BM->getDebugInfoEIS()), &EISId))
       return false;
   }
+  if (BM->preserveAllFunctionAttributesAndMetadata()) {
```

Comment:
What will happen if the 'preserve all' flag is set to true while `SPV_KHR_non_semantic_info` is explicitly disabled by `--spirv-ext=-SPV_KHR_non_semantic_info` . I'd expect it will result in an error. But do we want it to be an error? Should we also add a check here like `isAllowedToUseExtension(ExtensionID::SPV_KHR_non_semantic_info)`. Or should we enabled `SPV_KHR_non_semantic_info` implicitly by the flag?

Reply:
Just tried this:

```
build/tools/llvm-spirv/llvm-spirv --spirv-preserve-auxdata --spirv-ext=-SPV_KHR_non_semantic_info test/extensions/KHR/SPV_KHR_non_semantic_info/preserve-all-function-attributes.bc

RequiresExtension: Feature requires the following SPIR-V extension:
 SPV_KHR_non_semantic_info

```

imo implicitly enabling it makes sense, let me implement that, thanks

Reply:
ok i believe i implemented this in my latest commit, but im noticing something weird. if you pass `--spirv-preserve-auxdata` AND `---spirv-ext=-SPV_KHR_non_semantic_info`, it doesn't error and uses the extension anyway. is this what we want? i based my change on https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/3fe439333601c24d5e8f07862874a369854469be, maybe i did something dumb

Reply:
@MrSidims any take on the above? is the behavior incorrect?

Reply:
I didn't consider this corner case. I think if the extension is explicitly disallowed the translator should respect it. Hence setAllowedToUseExtension should be either modified or wrapped within another function, with finding the extension in the map and not changing Allowed status if the extension is found with 'false' flag. @svenvh WDYT?

Reply:
> I think if the extension is explicitly disallowed the translator should respect it. Hence setAllowedToUseExtension should be either modified or wrapped within another function, with finding the extension in the map and not changing Allowed status if the extension is found with 'false' flag. @svenvh WDYT?

I agree, that's the behavior I'd expect when the extension is explicitly disallowed.